### PR TITLE
Fix unexpected alb.Listener target group refresh

### LINKED ISSUE #3427: Root cause 3421 and revert patches 0042 and 0043
### What happened?
TLDR: `lb.Listener` panics if the `forward` parameter in `defaultActions` is not specified.

We shipped https://github.com/pulumi/pulumi-aws/pull/3426#issuecomment-1944127515 as a quick fix for https://github.com/pulumi/pulumi-aws/issues/3421

We should properly root cause the issue and revert the patches once we fix it.

More details in https://github.com/pulumi/pulumi-aws/issues/3421#issuecomment-1943569655

Note that the issue does not reproduce in TF and `PlanResourceChange` does not fix it https://github.com/pulumi/pulumi-aws/pull/3426#issuecomment-1944195346

### Example

`TestRegress3421` is a regression test for the issue - it should pass without the two upstream patches.

```typescript
import * as aws from "@pulumi/aws";

const vpc = new aws.ec2.Vpc("main", {cidrBlock: "10.0.0.0/16"});

const subnet = new aws.ec2.Subnet("main", {
    vpcId: vpc.id,
    cidrBlock: "10.0.12.0/24",
    availabilityZone: "us-east-1a"
});

const subnet2 = new aws.ec2.Subnet("subnet2", {
    vpcId: vpc.id,
    cidrBlock: "10.0.5.0/24",
    availabilityZone: "us-east-1b"
});

const secGroup = new aws.ec2.SecurityGroup("allowTls", {
    description: "Allow TLS inbound traffic and all outbound traffic",
    vpcId: vpc.id,
    tags: {
        Name: "allow_tls",
    },
});

const loadbalancer = new aws.lb.LoadBalancer("payload-lb", {
    loadBalancerType: "application",
    securityGroups: [secGroup.id],
    subnets: [subnet.id, subnet2.id],
    internal: true,
});

const targetGroup = new aws.lb.TargetGroup("payload-tg", {
port: 80,
protocol: "HTTP",
targetType: "ip",
vpcId: vpc.id,
});

new aws.lb.Listener("payload-lb-listner-http", {
loadBalancerArn: loadbalancer.arn,
port: 80,
protocol: "HTTP",
defaultActions: [
    {
    type: "forward",
    targetGroupArn: targetGroup.arn,
    },
],
});
```

panic:

```
Updating (dev)

View in Browser (Ctrl+O): https://app.pulumi.com/venelin-pulumi-corp/lb_panic/dev/updates/42

     Type                 Name                     Status            Info
     Type                 Name                     Status                  Info
     Type                 Name                     Status                  Info
     pulumi:pulumi:Stack  lb_panic-dev             **failed**              1 er
 +   └─ aws:lb:Listener   payload-lb-listner-http  **creating failed**     1 er

Diagnostics:
  aws:lb:Listener (payload-lb-listner-http):
    error: error reading from server: EOF

  pulumi:pulumi:Stack (lb_panic-dev):
    panic: value is null
    goroutine 32 [running]:
    github.com/hashicorp/go-cty/cty.Value.LengthInt({{{0x115bfd8a8?, 0x14004184880?}}, {0x0?, 0x0?}})
    	/home/runner/go/pkg/mod/github.com/hashicorp/go-cty@v1.4.1-0.20200723130312-85980079f637/cty/value_ops.go:989 +0x254
    github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbForwardActionOneOf({{{0x115bfd8a8?, 0x140041848b0?}}, {0x112e84260?, 0x14004100a20?}}, 0x110dde6e7?, {{0x14003340980, 0x7}, 0x0, 0x0, 0x0, ...}, ...)
    	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/service/elbv2/listener.go:1036 +0x18c
    github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbForwardAction(0x113cba6e0?, {0x110dde6e7, 0xe}, 0x5?, {{0x14003340980, 0x7}, 0x0, 0x0, 0x0, 0x140043124b0, ...}, ...)
    	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/service/elbv2/listener.go:1014 +0x188
    github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbListenerActions(0x112eb1480?, {0x110dde6e7, 0xe}, {0x14003346b40?, 0x1, 0x0?})
    	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/service/elbv2/listener.go:979 +0x240
    github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.resourceListenerRead({0x115bfd528, 0x140041bcae0}, 0x14004182600, {0x1159ed3c0?, 0x140005f51d0?})
    	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/service/elbv2/listener.go:535 +0x508
    github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.resourceListenerCreate({0x115bfd528?, 0x140041bcae0}, 0x14004182600, {0x1159ed3c0?, 0x140005f51d0})
    	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/service/elbv2/listener.go:506 +0xe20
    github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0x0?, {0x1159ed3c0?, 0x140005f51d0?})
    	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/provider/intercept.go:112 +0x1d4
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x115bfd528?, {0x115bfd528?, 0x14004167860?}, 0xd?, {0x1159ed3c0?, 0x140005f51d0?})
    	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20240202163305-e2a20ae13ef9/helper/schema/resource.go:778 +0x64
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x1400214ed20, {0x115bfd528, 0x14004167860}, 0x0, 0x14004182380, {0x1159ed3c0, 0x140005f51d0})
    	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20240202163305-e2a20ae13ef9/helper/schema/resource.go:914 +0x86c
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2.v2Provider.Apply({0x14000147da0?, {0x14001fcf5a0?, 0x0?, 0x14004167ad0?}}, {0x115bfd528, 0x14004167860}, {0x110de90e5, 0xf}, {0x0?, 0x0}, ...)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.74.0/pkg/tfshim/sdk-v2/provider.go:122 +0x140
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2.(*providerWithPlanResourceChangeDispatch).Apply(0x14002183f50, {0x115bfd528, 0x14004167860}, {0x110de90e5, 0xf}, {0x0, 0x0}, {0x115c0e4e0, 0x14004182380})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.74.0/pkg/tfshim/sdk-v2/provider2.go:700 +0xc8
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.(*Provider).Create(0x14000c9f980, {0x115bfd528?, 0x14004167410?}, 0x14004093450)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.74.0/pkg/tfbridge/provider.go:1025 +0x654
    github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).Create.func1({0x115c5ba20?, 0x14000c9f980?})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7/muxer.go:363 +0x3c
    github.com/pulumi/pulumi-terraform-bridge/x/muxer.resourceMethod[...](0x14002baf020?, 0x14002b196c8, 0x14002b196a8?)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7/muxer.go:322 +0xc4
    github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).Create(0x14002b19708?, {0x115bfd528?, 0x14004167410?}, 0x113277320?)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7/muxer.go:362 +0x5c
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Create_Handler.func1({0x115bfd528, 0x14004167410}, {0x11520a260?, 0x14004093450})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.105.0/proto/go/provider_grpc.pb.go:593 +0x74
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x115bfd528, 0x140041670b0}, {0x11520a260, 0x14004093450}, 0x140040fb660, 0x14004100840)
    	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x2e8
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Create_Handler({0x11583a4e0?, 0x14002baf020}, {0x115bfd528, 0x140041670b0}, 0x14003fc3d80, 0x14002ac7c80)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.105.0/proto/go/provider_grpc.pb.go:595 +0x12c
    google.golang.org/grpc.(*Server).processUnaryRPC(0x14002ab2200, {0x115bfd528, 0x14004167020}, {0x115c2f300, 0x14001890820}, 0x14004168ea0, 0x14002c6b650, 0x11f0b1a40, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.61.0/server.go:1385 +0xb8c
    google.golang.org/grpc.(*Server).handleStream(0x14002ab2200, {0x115c2f300, 0x14001890820}, 0x14004168ea0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.61.0/server.go:1796 +0xc4c
    google.golang.org/grpc.(*Server).serveStreams.func2.1()
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.61.0/server.go:1029 +0x8c
    created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 56
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.61.0/server.go:1040 +0x150

    error: update failed

Resources:
    7 unchanged

Duration: 7s

```

### Output of `pulumi about`

.

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### LINKED ISSUE #3722: PATCH Revert-Merge-pull-request-35678-from-hashicorp-b-elb.patch
Revert-Merge-pull-request-35678-from-hashicorp-b-elb.patch

This workarounds a bridge bug temporarily

### LINKED ISSUE #3723: PATCH 0043-Revert-Merge-pull-request-35671-from-hashicorp-b-lb-.patch
0043-Revert-Merge-pull-request-35671-from-hashicorp-b-lb-.patch

This also reverts a bridge bug temporarily.

### LINKED ISSUE #4079: ALB listener refresh issue
### Describe what happened

If you create an ALB listener that points at two target groups:

```
const listener = new aws.lb.Listener("listener", {
    loadBalancerArn: lb.arn,
    port: 80,
    defaultActions: [{
        type: "forward",
        forward: {
            targetGroups: [
              {
                arn: tg.arn,
                weight: 1,
              },
              {
                arn: tg2.arn,
                weight: 2,
              },
            ],
          },
    }]
});
```
in the state the inputs and outputs look like this:

<details><summary>click to view state</summary>

```json
"inputs": {
    "__defaults": [],
        "defaultActions": [
            {
                "__defaults": [],
                "forward": {
                    "__defaults": [],
                    "targetGroups": [
                        {
                            "__defaults": [],
                            "arn": "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/tg-437c643/2cc5dc4aa6679775",
                            "weight": 1
                        },
                        {
                            "__defaults": [],
                            "arn": "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/tg2-4708d4f/51e9d2aaf7e04ae1",
                            "weight": 2
                        }
                    ]
                },
                "type": "forward"
            }
        ],
        "loadBalancerArn": "arn:aws:elasticloadbalancing:us-east-1:052848974346:loadbalancer/app/lb-017bb48/5ce3a2c632531cbf",
        "port": 80
        },
"outputs": {
        "__meta": "{\"e2bfb730-ecaa-11e6-8f88-34363bc7c4c0\":{\"create\":300000000000,\"update\":300000000000}}",
        "arn": "arn:aws:elasticloadbalancing:us-east-1:052848974346:listener/app/lb-017bb48/5ce3a2c632531cbf/864700f68bd28b6e",
        "defaultActions": [
            {
                "authenticateCognito": null,
                "authenticateOidc": null,
                "fixedResponse": null,
                "forward": {
                    "stickiness": {
                        "duration": 0,
                        "enabled": false
                    },
                    "targetGroups": [
                        {
                            "arn": "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/tg2-4708d4f/51e9d2aaf7e04ae1",
                            "weight": 2
                        },
                        {
                            "arn": "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/tg-437c643/2cc5dc4aa6679775",
                            "weight": 1
                        }
                    ]
                },
                "order": 1,
                "redirect": null,
                "targetGroupArn": "",
                "type": "forward"
            }
        ],
        "id": "arn:aws:elasticloadbalancing:us-east-1:1234567890:listener/app/lb-017bb48/5ce3a2c632531cbf/864700f68bd28b6e",
        "loadBalancerArn": "arn:aws:elasticloadbalancing:us-east-1:1234567890:loadbalancer/app/lb-017bb48/5ce3a2c632531cbf",
        "mutualAuthentication": null,
        "port": 80,
        "protocol": "HTTP",
        "sslPolicy": "",
        "tags": {},
        "tagsAll": {}
}            
```
</details>

which is what we would expect.

Now if I simulate CodeDeploy going from a blue/green to a single target, and delete one of the targets and then do a refresh, I get this diff:

<details><summary>Click to show diff</summary>

```
~ aws:lb/listener:Listener: (update)
        [id=arn:aws:elasticloadbalancing:us-east-1:1234567890:listener/app/lb-017bb48/5ce3a2c632531cbf/864700f68bd28b6e]
        [urn=urn:pulumi:dev::elb-ignore-changes-bug::aws:lb/listener:Listener::listener]
        [provider=urn:pulumi:dev::elb-ignore-changes-bug::pulumi:providers:aws::default_6_39_0::dc749d79-d458-4525-aff6-2371787d1a20]
      ~ defaultActions: [
          ~ [0]: {
                  - forward       : {
                      - stickiness  : {
                          - duration: 0
                          - enabled : false
                        }
                      - targetGroups: [
                      -     [0]: {
                              - arn   : "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/tg2-4708d4f/51e9d2aaf7e04ae1"
                              - weight: 2
                            }
                      -     [1]: {
                              - arn   : "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/tg-437c643/2cc5dc4aa6679775"
                              - weight: 1
                            }
                        ]
                    }
                  + targetGroupArn: "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/tg-437c643/2cc5dc4aa6679775"
                }
        ]
```
</details>

In other words it deletes the forwards and sets the `targetGroupArn` input. Which feels odd because it means that it's harder to control if you have the forwards changing target groups.

If we do the same with the upstream provider (specifically 5.52.0 because that's the version that v6.39.0 of the AWS provider is using) then that state looks like this:

<details><summary>State from the upstream provider</summary>
```json
"default_action": [
              {
                "authenticate_cognito": [],
                "authenticate_oidc": [],
                "fixed_response": [],
                "forward": [
                  {
                    "stickiness": [
                      {
                        "duration": 3600,
                        "enabled": false
                      }
                    ],
                    "target_group": [
                      {
                        "arn": "arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/tg-437c643/2cc5dc4aa6679775",
                        "weight": 1
                      }
                    ]
                  }
                ],
                "order": 1,
                "redirect": [],
                "target_group_arn": "",
                "type": "forward"
              }
            ],
```
</details>

So it looks like we're doing something different.

### Sample program

You can use the Pulumi program here: https://github.com/pierskarsenbarg/elb-ignore-changes-refresh

If you want to see the difference in the TF provider then you can use the above program to create the load balancer and target groups and then copy and paste the ARNs into this program:

```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "5.52.0"
    }
  }
}

provider "aws" {
  region = "us-east-1"
}

resource "aws_lb_listener" "listener" {
    load_balancer_arn = ""
    port = 81
    default_action {
      type = "forward"
      forward {
        target_group {
            arn = ""
            weight = 1
        }
        target_group {
          arn = ""
          weight = 1
        }
      }
    }
}
```

### Log output

Refresh logs here: https://gist.github.com/pierskarsenbarg/10832bc3750a81b791eac570379016fe

### Affected Resource(s)

_No response_

### Output of `pulumi about`

```
CLI
Version      3.120.0
Go Version   go1.22.4
Go Compiler  gc

Plugins
KIND      NAME    VERSION
resource  aws     6.39.0
resource  awsx    2.11.0
resource  docker  4.5.4
resource  docker  3.6.1
language  nodejs  unknown

Host
OS       darwin
Version  14.5
Arch     arm64

This project is written in nodejs: executable='/Users/piers/.nvm/versions/node/v20.11.1/bin/node' version='v20.11.1'

Current Stack: pierskarsenbarg/elb-ignore-changes-bug/dev

TYPE                                                 URN
pulumi:pulumi:Stack                                  urn:pulumi:dev::elb-ignore-changes-bug::pulumi:pulumi:Stack::elb-ignore-changes-bug-dev
pulumi:providers:awsx                                urn:pulumi:dev::elb-ignore-changes-bug::pulumi:providers:awsx::default_2_11_0
awsx:ec2:Vpc                                         urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc::lb-vpc
pulumi:providers:aws                                 urn:pulumi:dev::elb-ignore-changes-bug::pulumi:providers:aws::default_6_37_1
aws:ec2/vpc:Vpc                                      urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc::lb-vpc
aws:ec2/subnet:Subnet                                urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::lb-vpc-ecs-public-subnet-2
aws:ec2/internetGateway:InternetGateway              urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/internetGateway:InternetGateway::lb-vpc
aws:ec2/subnet:Subnet                                urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::lb-vpc-ecs-public-subnet-1
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::lb-vpc-ecs-public-subnet-2
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::lb-vpc-ecs-public-subnet-1
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::lb-vpc-ecs-public-subnet-2
aws:ec2/route:Route                                  urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::lb-vpc-ecs-public-subnet-2
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::lb-vpc-ecs-public-subnet-1
aws:ec2/route:Route                                  urn:pulumi:dev::elb-ignore-changes-bug::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::lb-vpc-ecs-public-subnet-1
pulumi:providers:pulumi                              urn:pulumi:dev::elb-ignore-changes-bug::pulumi:providers:pulumi::default
pulumi:providers:aws                                 urn:pulumi:dev::elb-ignore-changes-bug::pulumi:providers:aws::default_6_39_0
aws:ec2/securityGroup:SecurityGroup                  urn:pulumi:dev::elb-ignore-changes-bug::aws:ec2/securityGroup:SecurityGroup::lbSg
aws:lb/loadBalancer:LoadBalancer                     urn:pulumi:dev::elb-ignore-changes-bug::aws:lb/loadBalancer:LoadBalancer::lb
aws:lb/targetGroup:TargetGroup                       urn:pulumi:dev::elb-ignore-changes-bug::aws:lb/targetGroup:TargetGroup::tg
aws:lb/targetGroup:TargetGroup                       urn:pulumi:dev::elb-ignore-changes-bug::aws:lb/targetGroup:TargetGroup::tg2
aws:lb/listener:Listener                             urn:pulumi:dev::elb-ignore-changes-bug::aws:lb/listener:Listener::listener


Found no pending operations associated with dev

Backend
Name           pulumi.com
URL            https://app.pulumi.com/pierskarsenbarg
User           pierskarsenbarg
Organizations  pierskarsenbarg, karsenbarg, team-ce, gitlab-test-piers, demo
Token type     personal

Dependencies:
NAME            VERSION
@pulumi/pulumi  3.119.0
@types/node     18.19.34
typescript      5.4.5
@pulumi/aws     6.39.0
@pulumi/awsx    2.11.0

Pulumi locates its logs in /var/folders/x8/cdd9j87s607fwpy0q62mfmmw0000gn/T/ by default
```

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by t0yv0**
Fixes unexpected alb.Listener target group refresh.

The change was fixed upstream but patched out due to an unrelated problem causing a panic in Pulumi. The root cause is now fixed in the bridge. This change drops the two obsolete patches to catch up with upstream, and enrolls alb.Listener in PlanResourceChange version of the bridge that fixes the issue.

Fixes #3427
Fixes #3722 
Fixes #3723
Fixes #4079

### CODE REVIEW

**Comment by t0yv0** [comment]
@VenelinMartinov there's a lot of detail in https://github.com/pulumi/pulumi-aws/issues/3427 but it seems that the https://github.com/pulumi/pulumi-aws/issues/3723 patch is already covered by tests so reverting it should be safe?

```
--- PASS: TestRegress3421Update (232.17s)
--- PASS: TestRegress3421 (285.27s)
```

Double checking, if you think a test still needs to be added I can do that.

**Comment by t0yv0** [comment]
The other patch cannot be dropped yet but can be dropped on the next bridge release:

https://github.com/pulumi/pulumi-aws/issues/3722

**Comment by t0yv0** [comment]
Suspect https://github.com/pulumi/pulumi-aws/issues/3722 with bridge upgrade as described there would fix this test.

**Review by mikhailshilkov — APPROVED** [review]
LGTM

File: provider/pkg/elb/elb.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func configureClient() *elbv2.Client {
+	loadOpts := []func(*config.LoadOptions) error{}
+	if p, ok := os.LookupEnv("AWS_PROFILE"); ok {
```
Comment (VenelinMartinov) [inline provider/pkg/elb/elb.go:49]: Was this copied from somewhere? Is it missing some other options around AWS config? Like logging in with a token directly etc?

**Review by VenelinMartinov — APPROVED** [review]
looks good, just a question around interacting with the AWS sdk

File: provider/pkg/elb/elb.go (hunk 1 above)
Comment (t0yv0) [reply provider/pkg/elb/elb.go:49]: Copilot I think. This is only used for testing AFAIK. Should it perhaps live in a foo_test.go file?

Comment (VenelinMartinov) [reply provider/pkg/elb/elb.go:49]: Ah, that's fine then, yeah, could be a bit clearer that it's test only but not a big deal

**Pull request merged**